### DISCUSSION TITLE
.gitattributes linguist settings incorrectly formatted?

### DISCUSSION THREAD

**Discussion by quoid** [github-linguist/linguist#5719] [Q&A] [answered]
One of [my projects](https://github.com/quoid/userscripts) shows up as mostly `html` which I reckon is due to large generated html files. 

[Here's a link to my .gitattibutes file](https://github.com/quoid/userscripts/blob/master/.gitattributes).

I am curious about a few things in particular:

- does my `.gitattributes` file look properly formatted? (it seems like certain `html` files are not being ignored)
- what is the proper way to handle spaces in file paths?
  - [this thread](https://github.com/github/linguist/discussions/5491) suggests using `[[:space:]]` or `\`
  - [this issue](https://github.com/github/linguist/issues/5117) suggests using `?` or `*`
- is a `-` needed when flagging individual files?
  - all individual files in the [override doc](https://github.com/github/linguist/blob/master/docs/overrides.md) have a `-` before the attribute
- is there a way to ignore all files of a certain type that are contained in a directory and that directory's sub directories? something like `folder/subfolder/**/*.js`?

Thank you in advance for any help you can provide

**Comment by lildude** [comment, marked as answer]
> * does my `.gitattributes` file look properly formatted? (it seems like certain `html` files are not being ignored)

It does now 😁. I see you've worked out that the `-` at the beginning negates the option. Your earlier versions of the file were wrong if you were intending on having the HTML files ignored.

> what is the proper way to handle spaces in file paths?
> 
>     * [this thread](https://github.com/github/linguist/discussions/5491) suggests using `[[:space:]]` or `\`
> 
>     * [this issue](https://github.com/github/linguist/issues/5117) suggests using `?` or `*`

Both essentially work as `[[:space:]]` matches the specific space "character" whilst `*` and `?` match any character, including spaces.

> * is a `-` needed when flagging individual files?

No. As you've worked out, this is about negation, not specific files.

> * is there a way to ignore all files of a certain type that are contained in a directory and that directory's sub directories? something like `folder/subfolder/**/*.js`?

Yes. Exactly like this.

The [gitattributes](https://git-scm.com/docs/gitattributes) docs have a lot more info on how to use the gitattributes file.

**Reply by quoid** [reply]
I did get it working, @lildude but it's still great info you shared. Thanks!
